doc/user/content/sql/create-source/webhook.md

```diff
@@ -0,0 +1,108 @@
+`HEADERS`              | `map[text=>text]` or `map[text=>bytea]` | Provide a column `'headers'` to the check expression. The column can be renamed with the optional **AS** _alias_ statement, and the type can be changed to `map[text => bytea]` with the optional **BYTES** keyword.
+`SECRET` _secret_name_ | `text` or `bytea`    | Provide a column _secret_name_ to the check expression, with the value of the [`SECRET`](/sql/create-secret) _secret_name_. The column can be renamed with the optional **AS** _alias_ statement, and the type can be changed to `bytea` with the optional **BYTES** keyword.
+
+## Webhook URL
+
+After creating a webhook source, requests can be sent to `https://<HOST>/api/webhook/<database>/<schema>/<src_name>`.
+Where `<HOST>` is the URL for your Materialize instance, which can be found on the [Materialize Web
+Console](https://console.materialize.com/). Then `<database>` and `<schema>` are the database and
+schema where you created your source, and `<src_name>` is the name you provided for your source at
+the time of creation.
+
+
+> **_Note:_** This is a public URL that is open to the internet and has no security. To validate that
+requests are legitimate see Validating Requests. For limits imposed on this endpoint see Request Limits.
+
+## Validating Requests
+
+It's common for applications using webhooks to provide a method for validating a request is
+legitimate. Using `CHECK` you can specify an expression to do this validation for your Webhook
+Source.
+
+For example, the following source HMACs the request body using the SHA256 hashing algorithm, and
+asserts the result is equal to the value provided in the `x-signature` header, decoded with base64.
+
+```
+CREATE SOURCE my_webhook_source IN CLUSTER my_cluster FROM WEBHOOK
+  BODY FORMAT JSON
+  CHECK (
+    WITH (
+      HEADERS, BODY AS request_body,
+      SECRET my_webhook_shared_secret,
+    )
+    decode(headers->'x-signature', 'base64') = hmac(request_body, my_webhook_shared_secret, 'sha256')
+  )
+```
+
+The body and headers of the request are only provided for validation if `WITH ( BODY, HEADERS, ... )`
+is specified as part of the `CHECK` statement. By default the type of `body` used for validation is
+`text`, regardless of the `BODY FORMAT` you specified for the source. In the example above, the
+`body` column for the `my_webhook_source` has a type of `jsonb`, but `request_body` as used in the
+validation expression has type `text`. Futher, the request headers are not persisted as part of
+`my_webhook_source` since `INCLUDE HEADERS` was not specified, but they are provided to the
+validation expression.
+
+
+> **_Note:_** Without a `CHECK` statement **all requests will be accepted**. To prevent bad actors from
+inserting data it is **strongly encouraged** that you define a `CHECK` statement with your webhook
+sources.
```

Comment:
```suggestion
{{< callout >}}
> **_Note:_** Without a `CHECK` statement **all requests will be accepted**. To prevent bad actors from
inserting data it is **strongly encouraged** that you define a `CHECK` statement with your webhook
sources.
{{< /callout >}}
```